### PR TITLE
enforce view_report permission more strictly

### PR DESCRIPTION
**Pull request by sravfeyn**
https://dimagi-dev.atlassian.net/browse/HI-435

If a user has 'Edit Data' permission but not 'View Reports' permission, the data export downloads page (which 403s when clicked on download already) should 403 as well. The default landing page for Data tab for such user is updated to 'Reassign Cases' which the user has access to because of 'Edit Data'

### CODE REVIEW

**Comment by orangejenny** [comment]
fyi @biyeun in case this overlaps with your permissions work

**Comment by esoergel** [comment]
Does anything need to change in this class? 
https://github.com/dimagi/commcare-hq/blob/73f08b5da1b4eaa4cf1f804830c780d96742c9ff/corehq/apps/export/views/utils.py#L65-L65

**Comment by calellowitz** [comment]
What if the user has permission to view reports but not to edit data?

**Review by biyeun — APPROVED** [review]

**Comment by biyeun** [comment]
@orangejenny looks good as far as overlap with work i've done. thanks for flagging

**Comment by sravfeyn** [comment]
@calellowitz, Good catch, I updated so that viewing/download exports requires view_report permission (regardless of edit_data)

@esoergel Thanks, I updated `ExportPermissionManager` and reused it in the methods I modified, so that it's consistent everywhere.

File: corehq/apps/export/views/utils.py [hunk 1]
```diff
@@ -311,3 +311,11 @@ def delete(self, request, *args, **kwargs):
...
         data_file.delete()
         return HttpResponse(status=204)
+
+
+def can_view_form_exports(couch_user, domain):
+    return ExportsPermissionsManager('form', domain, couch_user).has_form_export_permissions
+
+
+def can_view_case_exports(couch_user, domain):
+    return ExportsPermissionsManager('case', domain, couch_user).has_form_export_permissions
```
Comment (esoergel) [inline corehq/apps/export/views/utils.py]: :+1:  I was gonna propose moving these.

**Review by esoergel — APPROVED** [review]
Nice job consolidating this stuff

**Pull request merged**